src/app/shared/services/authentication.service.ts

```diff
@@ -4,18 +4,73 @@ import 'firebase/analytics';
 import 'firebase/auth';
 import 'firebase/firestore';
 import {firebaseConfig} from '../../../firebase-config';
+import {Router} from '@angular/router';
+import {NotifierService} from '../components/notifier/notifier.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
 
-  constructor() {
+  provider = new firebase.auth.GoogleAuthProvider();
+
+  constructor(private router: Router,
+              private notifierService: NotifierService) {
     firebase.initializeApp(firebaseConfig);
   }
 
   signUpWithEmailAndPassword(email, password) {
-    firebase.auth().createUserWithEmailAndPassword(email, password).catch((error) => console.log(error)).then(res => console.log(res));
+    firebase.auth().createUserWithEmailAndPassword(email, password).catch((error) => {
+      if (error) {
+        this.notifierService.notify(
+          error.message,
+          2,
+        );
+      }
+    }).then(res => {
+      if (res) {
+        this.redirectToDashboard();
+        console.log(res);
+      }
+    });
+
+  }
+
+  signInWithEmailAndPassword(email, password) {
+    firebase.auth().signInWithEmailAndPassword(email, password)
+      .catch((error) => console.log(error))
+      .then(res => {
+        this.redirectToDashboard();
+        console.log(res);
+      });
+  }
+
+  signInWithGoogle() {
+    firebase.auth().signInWithRedirect(this.provider).then((res) => {
+      console.log(res);
+      firebase.auth().getRedirectResult().then((result) => {
+        console.log(result);
+      }).catch((error) => {
+        console.log(error);
+      });
+    });
+
+  }
+
+  signOut() {
+    firebase.auth().signOut().then(res => console.log(res));
+  }
+
+  isAuthenticated(): boolean {
+    const currentUser = firebase.auth().currentUser;
+    return !!currentUser;
+  }
 
+  redirectToDashboard() {
+    firebase.auth().onAuthStateChanged((user) => {
```

Comment:
The "onAuthStateChanged" listener is not needed here. Function is being used only in places where user is already present (after successful signin/signup), so there is no reason to check it again.